webrtc.go

```diff
@@ -105,17 +129,64 @@ func newSession(config SessionConfig) (*Session, error) {
 		scopedLogger.Warn().Err(err).Msg("Failed to create PeerConnection")
 		return nil, err
 	}
+
 	session := &Session{peerConnection: peerConnection}
 	session.rpcQueue = make(chan webrtc.DataChannelMessage, 256)
+	session.initQueues()
+
 	go func() {
 		for msg := range session.rpcQueue {
-			onRPCMessage(msg, session)
+			// TODO: only use goroutine if the task is asynchronous
+			go onRPCMessage(msg, session)
 		}
 	}()
 
+	for i := 0; i < len(session.hidQueue); i++ {
+		go session.handleQueues(i)
+	}
+
 	peerConnection.OnDataChannel(func(d *webrtc.DataChannel) {
+		defer func() {
+			if r := recover(); r != nil {
+				scopedLogger.Error().Interface("error", r).Msg("Recovered from panic in DataChannel handler")
+			}
+		}()
+
 		scopedLogger.Info().Str("label", d.Label()).Uint16("id", *d.ID()).Msg("New DataChannel")
+
 		switch d.Label() {
+		case "hidrpc":
+			session.HidChannel = d
+			d.OnMessage(func(msg webrtc.DataChannelMessage) {
+				l := scopedLogger.With().Str("data", string(msg.Data)).Int("length", len(msg.Data)).Logger()
```

Comment:
The call to `string(msg.Data)` will execute here and allocate/copy the data to a string even if we subsequently do not call any `.Warn()` or `.Trace()` (and at trace logging level). If we use .Bytes("data", msg.Data) like we do in [_jsonrpc_](https://github.com/jetkvm/kvm/blob/dev/jsonrpc.go#L414)